### PR TITLE
[blog] Adding Introducing Slack Integration for Hue blog post

### PR DESCRIPTION
**Pull request by Harshg999**
## What changes were proposed in this pull request?

- Blog post for v1 Slack App

## To-Do
- Update image links after all the image are uploaded to CDN
- Update security section with example (after discussion)

### CODE REVIEW

File: docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+Paste it in the Slack channel and see the unfurl magic!
+
+![Query Link Preview]()
+
+### Wonder why the query result (if available) is shown in this way?
+
+Slack currently does not support markdown tables and potential improvements with inline preview will come when Hue supports result caching via [query tasks](https://docs.gethue.com/administrator/administration/reference/#task-server)
+
+After evaluating a lot of possible fixes ( like uploading result image, truncating columns which doesn't look good, pivoting table, uploading result file etc.) and seeing their tradeoffs, we chose to have few sample rows but keep all columns by pivoting the result table and to compensate for the loss of rows, Hue app gives the result file in the message thread.
```
Comment (romainr) [inline docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md:78]: `uploading result file`, could link to the slack issue you open

File: docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+![Message Thread with Result File]()
+
+Users can share the SQL gists too!
+
+![Gist Link]()
+
+![Gist Link Preview]()
+
+Keeping in mind the security aspect, those Slack users who are Hue users and have the read permissions to access the query and its result will get this rich preview and result file after sharing the link. This mapping is currently done by checking the email prefix for Hue username.
```
Comment (romainr) [inline docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md:88]: Would give an example of mapping

File: docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+</br>
+</br>
+
+Any feedback or question? Feel free to comment here or on the <a href="https://discourse.gethue.com/">Forum</a> and <a href="https://docs.gethue.com/quickstart/">quick start</a> SQL querying!
+
+
+Onwards!
+
+~ Harsh and Romain from the Hue Team
```
Comment (romainr) [inline docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md:106]: Just put Harsh!

**Review by romainr — APPROVED** [review]
:+1: with the links !

File: docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md [hunk 4]
```diff
@@ -57,33 +57,33 @@ What could be better than improving the collaboration by using one of the best c
 
 Although currently in **beta**, this app integration expands the ability to share query links or gists to the desired Slack channels which then unfurls in a rich preview for other users. It even gives you the result file if the result has not expired.
 
-![Share to Slack Architecture]()
+![Share to Slack Architecture](https://cdn.gethue.com/uploads/2021/04/share_to_slack_architecture.png)
 
 The main bot logic lies on top of the Hue server listening to the events posted by the Hue App from Slack to an endpoint, processing the data accordingly and using Slack API methods for sending the desired response.
 
 To manually set-up the Hue App, follow the steps mentioned in the [link here]()
```
Comment (romainr) [inline docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md:64]: link

File: docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md [hunk 5]
```diff
@@ -57,33 +57,33 @@ What could be better than improving the collaboration by using one of the best c
...
 
 Now, open Hue, run some query and copy its link
```
Comment (romainr) [inline docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md:66]: end of line :
?

File: docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md [hunk 6]
```diff
@@ -57,33 +57,33 @@ What could be better than improving the collaboration by using one of the best c
...
 
-![Run Query in Hue]()
+![Run Query in Hue](https://cdn.gethue.com/uploads/2021/04/run_query_in_hue.png)
 
 Paste it in the Slack channel and see the unfurl magic!
 
-![Query Link Preview]()
+![Query Link Preview](https://cdn.gethue.com/uploads/2021/04/query_link_preview.png)
 
 ### Wonder why the query result (if available) is shown in this way?
```
Comment (romainr) [inline docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md:74]: FYI: pretty long title (does not always show up well)

File: docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md (hunk 2 above)
Comment (romainr) [reply docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md:88]: same

File: docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md (hunk 6 above)
Comment (Harshg999) [reply docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md]: Reduced the heading a bit

File: docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md (hunk 2 above)
Comment (Harshg999) [reply docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md]: Mapping example given @romainr

File: docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+**And when there is Slack, there are Slack Apps!** And our beloved Hue is getting one!
+
+Although currently in **beta**, this app integration expands the ability to share query links or gists to the desired Slack channels which then unfurls in a rich preview for other users. It even gives you the result file if the result has not expired.
+
+![Share to Slack Architecture](https://cdn.gethue.com/uploads/2021/04/share_to_slack_architecture.png)
+
+The main bot logic lies on top of the Hue server listening to the events posted by the Hue App from Slack to an endpoint, processing the data accordingly and using Slack API methods for sending the desired response.
+
+To manually set-up the Hue App, follow the steps mentioned in the [link here](https://docs.gethue.com/administrator/configuration/server/#manual-slack-app-installation)
```
Comment (romainr) [inline docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md:64]: `link here`

File: docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+Paste it in the Slack channel and see the unfurl magic!
+
+![Query Link Preview](https://cdn.gethue.com/uploads/2021/04/query_link_preview.png)
+
+### Wonder why result is shown like this?
+
+Slack currently does not support markdown tables and potential improvements with inline preview will come when Hue supports result caching via [query tasks](https://docs.gethue.com/administrator/administration/reference/#task-server).
+
+After evaluating a lot of possible fixes ( like uploading result image, truncating columns which doesn't look good, pivoting table, [uploading result file](https://github.com/slackapi/python-slack-sdk/issues/991) etc.) and seeing their tradeoffs, we chose to have few sample rows but keep all columns by pivoting the result table and to compensate for the loss of rows, Hue app gives the result file in the message thread.
```
Comment (romainr) [inline docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md:78]: extra space before `like`

File: docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+![Gist Link](https://cdn.gethue.com/uploads/2021/04/gist_link.png)
+
+![Gist Link Preview](https://cdn.gethue.com/uploads/2021/04/gist_link_preview.png)
+
+### Read access
+
+Keeping in mind the security aspect, those Slack users who are Hue users and have the read permissions to access the query and its result will get this rich preview and result file after sharing the link. This mapping is currently done by checking the email prefix and its host for Hue username.
+
+For example, some person ‘Alice’ having a Hue account with username ‘alice’ can have read access from some Slack account only if the email prefix of that slack user is same and Hue username and the email host is same in the Hue domain i.e. **alice@gethue.com slack user** can only access **Hue user ‘alice’** on **‘demo.gethue.com’**.
```
Comment (romainr) [inline docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md:92]: demo.gethue.com --> hue.gethue.com

might be simpler to understand

File: docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+
+### What's coming up next?
+
+In the follow-ups, promoting the Share to Slack from the Hue side where users can select the channel to send the link.
+
+Also stay tuned for more such functionalities such as replying to users asking questions on how to find certain data tables or to query them and also send results for scheduled queries in certain Slack channels!
+
+Till then, run some queries, copy those links and share ‘em all!
+
+</br>
```
Comment (romainr) [inline docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md:102]: Wondering: 
for a 1 click try...
add the demo slack URL here and demo.gethue links here and demo credentials?

**Review by romainr — APPROVED** [review]
Nice!

File: docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md (hunk 10 above)
Comment (Harshg999) [reply docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md:114]: Added the `Try it out!` sub-section in the blog

File: docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+
+Also stay tuned for more such functionalities such as replying to users asking questions on how to find certain data tables or to query them and also send results for scheduled queries in certain Slack channels!
+
+### Try it out!
+
+Get your hands on this beta feature by:
+
+Running some queries on the [demo live instance](https://demo.gethue.com/).
+
+Joining the [demo Slack workspace](https://join.slack.com/t/hue-bot-dev/shared_invite/zt-opqwvv68-eQFeios8FzFbmqQJ5wBNzg) to see the app in action or by using the following **demo Slack account credentials** to share some query links in the **demo workspace**.
```
Comment (romainr) [inline docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md:106]: `**demo Slack account credentials** to share some query links in the **demo workspace**.` --> credentials

**Review by romainr — APPROVED** [review]

File: docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+
+```
+Slack account email: demo@gethue.com
+Slack account password: demohuedemo
```
[the remark this answers is not in the document]
Comment (Harshg999) [reply docs/gethue/content/en/posts/2021-04-09-collaborate-on-your-sql-queries-and-results-directly-within-slack.md:110]: > why no just `demo`?

Min. 6 character password required for Slack account

**Comment by romainr** [comment]
ok so put 6 please!

**Pull request merged**